src/body/body.rs

```diff
@@ -332,6 +322,35 @@ impl Payload for Body {
     }
 }
 
+impl ::http_body::Body for Body {
+    type Data = Chunk;
+    type Error = ::Error;
+
+    fn poll_data(&mut self) -> Poll<Option<Self::Data>, Self::Error> {
+        <Self as Payload>::poll_data(self)
+    }
+
+    fn poll_trailers(&mut self) -> Poll<Option<HeaderMap>, Self::Error> {
+        <Self as Payload>::poll_trailers(self)
+    }
+
+    fn is_end_stream(&self) -> bool {
+        <Self as Payload>::is_end_stream(self)
+    }
+
+    fn size_hint(&self) -> SizeHint {
+        let mut hint = SizeHint::default();
+
+        let upper_limit = <Self as Payload>::content_length(self);
+
+        if let Some(upper) = upper_limit {
+            hint.set_upper(upper);
```

Comment:
I believe we can set the lower as well, since it's meant to be an exact size in `Payload`.

Reply:
so if I understand correctly, since we ahve the total content_length here so the exact size, lower would == upper?

Reply:
Yep, `Payload::content_length` is an exact size if `Some(_)`.